### PR TITLE
Vehicle config to avoid Issue 2677

### PR DESCRIPTION
**Pull request by schwiti6190**
- Fix for #2677 
- Added new vehicle config: "disableUnfoldBaleLoader" to disable bale loader unfold.
- New console commands for working with vehicle configurations:
  - Print entered vehicle and implement config xml filenames.
  - Print attribute for entered vehicle and implements.
  - Print all assigned attributes for entered vehicle and implements.
  - Print all valid attribute names
- All config commands are now under: "cpVehicleConfigurations..."

### CODE REVIEW

File: scripts/ai/controllers/FoldableController.lua [hunk 1]
```diff
@@ -0,0 +1,42 @@
+--[[
+This file is part of Courseplay (https://github.com/Courseplay/courseplay)
+Copyright (C) 2019-2022 Peter Vaiko
```
Comment (pvaiko) [inline scripts/ai/controllers/FoldableController.lua:3]: year, author

File: scripts/specializations/CpAIWorker.lua [hunk 2]
```diff
@@ -504,4 +505,33 @@ function CpAIWorker:onStartAutoDrive()
...
+    g_devHelper.consoleCommands:registerConsoleCommand("cpVehicleOnWorkStartTest", 
+        "Raise work start command", "consoleCommandRaiseWorkStart", CpAIWorker)
+    --- TODO: Add function to simulate lowering and raising with controllers.
+end
+
+--- Helper command to test the pipe measurement.
+function CpAIBaleFinder:consoleCommandRaiseWorkStart()
+    local vehicle = g_currentMission.controlledVehicle
+    if not vehicle then 
+        CpUtil.info("Could not measure pipe properties without entering a vehicle!")
```
Comment (pvaiko) [inline scripts/specializations/CpAIWorker.lua:524]: This probably isn't the correct log message.

File: scripts/specializations/CpAIWorker.lua [hunk 3]
```diff
@@ -504,4 +505,33 @@ function CpAIWorker:onStartAutoDrive()
...
+--- Console commands
+---------------------------------------------
+
+function CpAIWorker.registerConsoleCommands()
+    g_devHelper.consoleCommands:registerConsoleCommand("cpVehicleOnWorkStartTest", 
+        "Raise work start command", "consoleCommandRaiseWorkStart", CpAIWorker)
+    --- TODO: Add function to simulate lowering and raising with controllers.
+end
+
+--- Helper command to test the pipe measurement.
```
Comment (pvaiko) [inline scripts/specializations/CpAIWorker.lua:520]: pipe?

File: scripts/specializations/CpAIWorker.lua [hunk 4]
```diff
@@ -504,4 +505,33 @@ function CpAIWorker:onStartAutoDrive()
...
\ No newline at end of file
+end
+
+---------------------------------------------
+--- Console commands
+---------------------------------------------
+
+function CpAIWorker.registerConsoleCommands()
+    g_devHelper.consoleCommands:registerConsoleCommand("cpVehicleOnWorkStartTest", 
+        "Raise work start command", "consoleCommandRaiseWorkStart", CpAIWorker)
```
Comment (pvaiko) [inline scripts/specializations/CpAIWorker.lua:516]: It is a bit confusing what this is doing especially it seems to raise an event (not raise the implements) which lowers the implements. Also, only foldable controllers are notified, so its scope seems to be the foldables only, what about the other controllers?

**Review by pvaiko — CHANGES_REQUESTED** [review]

File: scripts/specializations/CpAIWorker.lua (hunk 4 above)
Comment (schwiti6190) [reply scripts/specializations/CpAIWorker.lua:516]: Forgot to comment my intentions there, was to acting to fast ...
For now I only needed the foldable controller for testing and the other could be added later when needed.

**Comment by Tensuko** [comment]
@schwiti6190 conflict here now too, also what is the status?

**Comment by Tensuko** [comment]
Rebased and added config and tested it with AutoLoad from Loki, does work.
Anything missing?

**Pull request merged**